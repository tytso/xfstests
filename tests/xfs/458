#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2018 Oracle, Inc.  All rights reserved.
#
# FS QA Test No. 458
#
# Populate a XFS filesystem and fuzz every bnobt key/pointer.
# Do not fix the filesystem, to test metadata verifiers.

. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_norepair

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

path="$(_scratch_xfs_find_agbtree_height 'bno' 2)" || \
	_fail "could not find two-level bnobt"

echo "Fuzz bnobt keyptr"
_scratch_xfs_fuzz_metadata '' 'none'  "$path" 'addr bnoroot' >> $seqres.full
echo "Done fuzzing bnobt keyptr"

# success, all done
status=0
exit
