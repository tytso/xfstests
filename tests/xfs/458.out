QA output created by 458
Format and populate
Fuzz bnobt keyptr
leftsib = add: offline scrub didn't fail.
rightsib = add: offline scrub didn't fail.
keys[1].startblock = zeroes: offline scrub didn't fail.
keys[1].startblock = ones: offline scrub didn't fail.
keys[1].startblock = firstbit: offline scrub didn't fail.
keys[1].startblock = middlebit: offline scrub didn't fail.
keys[1].startblock = lastbit: offline scrub didn't fail.
keys[1].startblock = add: offline scrub didn't fail.
keys[1].startblock = sub: offline scrub didn't fail.
keys[1].blockcount = zeroes: offline scrub didn't fail.
keys[1].blockcount = zeroes: online scrub didn't fail.
keys[1].blockcount = ones: offline scrub didn't fail.
keys[1].blockcount = ones: online scrub didn't fail.
keys[1].blockcount = firstbit: offline scrub didn't fail.
keys[1].blockcount = firstbit: online scrub didn't fail.
keys[1].blockcount = middlebit: offline scrub didn't fail.
keys[1].blockcount = middlebit: online scrub didn't fail.
keys[1].blockcount = lastbit: offline scrub didn't fail.
keys[1].blockcount = lastbit: online scrub didn't fail.
keys[1].blockcount = add: offline scrub didn't fail.
keys[1].blockcount = add: online scrub didn't fail.
keys[1].blockcount = sub: offline scrub didn't fail.
keys[1].blockcount = sub: online scrub didn't fail.
keys[2].startblock = zeroes: offline scrub didn't fail.
keys[2].startblock = ones: offline scrub didn't fail.
keys[2].startblock = firstbit: offline scrub didn't fail.
keys[2].startblock = middlebit: offline scrub didn't fail.
keys[2].startblock = lastbit: offline scrub didn't fail.
keys[2].startblock = add: offline scrub didn't fail.
keys[2].startblock = sub: offline scrub didn't fail.
keys[2].blockcount = zeroes: offline scrub didn't fail.
keys[2].blockcount = zeroes: online scrub didn't fail.
keys[2].blockcount = ones: offline scrub didn't fail.
keys[2].blockcount = ones: online scrub didn't fail.
keys[2].blockcount = firstbit: offline scrub didn't fail.
keys[2].blockcount = firstbit: online scrub didn't fail.
keys[2].blockcount = middlebit: offline scrub didn't fail.
keys[2].blockcount = middlebit: online scrub didn't fail.
keys[2].blockcount = lastbit: offline scrub didn't fail.
keys[2].blockcount = lastbit: online scrub didn't fail.
keys[2].blockcount = add: offline scrub didn't fail.
keys[2].blockcount = add: online scrub didn't fail.
keys[2].blockcount = sub: offline scrub didn't fail.
keys[2].blockcount = sub: online scrub didn't fail.
Done fuzzing bnobt keyptr
